diff --git a/lib/Cloudflare/Workers/DurableObjectNamespace.d.ts b/lib/Cloudflare/Workers/DurableObjectNamespace.d.ts
index 49dae40f64388d102199797458151f5c1f5c79aa..43cdc888695b449a51c146606e74636f69b96e37 100644
--- a/lib/Cloudflare/Workers/DurableObjectNamespace.d.ts
+++ b/lib/Cloudflare/Workers/DurableObjectNamespace.d.ts
@@ -53,7 +53,7 @@ export interface DurableObjectShape {
     webSocketMessage?: (socket: DurableWebSocket, message: string | ArrayBuffer) => Effect.Effect<void>;
     webSocketClose?: (socket: DurableWebSocket, code: number, reason: string, wasClean: boolean) => Effect.Effect<void>;
 }
-export type DurableObjectServices = DurableObjectNamespace | DurableObjectState | WorkerServices | WorkerEnvironment | PlatformServices;
+export type DurableObjectServices = DurableObjectNamespace | DurableObjectState | WorkerServices | WorkerEnvironment | PlatformServices | DurableObjectNamespaceScope;
 export interface DurableObjectNamespaceProps {
     /**
      * Name of the exported `DurableObject` class.
