diff --git a/lib/Cloudflare/Workers/InferEnv.d.ts b/lib/Cloudflare/Workers/InferEnv.d.ts
index 02a9b8d08d33a156cf2759b3947f862a74169806..f2b27a166c2a69a9c85b3897606657020f8b1078 100644
--- a/lib/Cloudflare/Workers/InferEnv.d.ts
+++ b/lib/Cloudflare/Workers/InferEnv.d.ts
@@ -5,7 +5,9 @@ import type { Rpc } from "../../Rpc.ts";
 import type * as Cloudflare from "../index.ts";
 import type { RpcErrorEnvelope, RpcStreamEnvelope } from "./Rpc.ts";
 import type { Worker } from "./Worker.ts";
-export type InferEnv<W> = W extends Effect.Effect<infer A, infer _E, infer _R> ? InferEnv<A> : W extends Worker<any> ? InferEnv<Exclude<W["Props"]["env"], undefined>> : {
+export type InferEnv<W> = W extends Effect.Effect<infer A, infer _E, infer _R> ? InferEnv<A> : W extends abstract new (...args: any) => infer I ? InferEnv<I> : W extends Worker<any> ? {
+    [k in keyof Exclude<W["Props"]["env"], undefined>]: GetBindingType<Exclude<W["Props"]["env"], undefined>[k]>;
+} : {
     [k in keyof W]: GetBindingType<W[k]>;
 };
 export type GetBindingType<T> = T extends Effect.Effect<infer A, infer _E, infer _R> ? GetBindingType<A> : T extends Cloudflare.Assets ? Service : T extends Rpc<infer Shape extends object> ? RpcWireShape<Shape> & Service : T extends Cloudflare.D1Database ? D1Database : T extends Cloudflare.R2Bucket ? R2Bucket : T extends Cloudflare.KVNamespace ? KVNamespace : T extends Cloudflare.Queue ? Queue<unknown> : T extends Cloudflare.AiGateway ? Ai : T extends Cloudflare.SendEmail ? SendEmail : T extends Cloudflare.AnalyticsEngineDataset ? AnalyticsEngineDataset : T extends Cloudflare.Artifacts ? Artifacts : T extends Cloudflare.Images ? ImagesBinding : T extends Cloudflare.BrowserRendering ? Fetcher : T extends Cloudflare.Hyperdrive ? Hyperdrive : T extends Cloudflare.DurableObjectNamespaceLike ? DurableObjectNamespace<Exclude<T["Shape"], undefined>> : T extends Redacted<any> ? string : T;
